### PR TITLE
Fix for large I/O

### PR DESCRIPTION
**Pull request by flaub**

### CODE REVIEW

File: risc0/zkvm/platform/src/syscall.rs [hunk 1]
```diff
@@ -71,6 +71,9 @@ pub mod reg_abi {
 
 pub const DIGEST_WORDS: usize = 8;
 pub const DIGEST_BYTES: usize = WORD_SIZE * DIGEST_WORDS;
+pub const MAX_BUF_BYTES: usize = 64 * 1024;
+pub const MAX_BUF_WORDS: usize = MAX_BUF_BYTES / WORD_SIZE;
+pub const MAX_SHA_COMPRESS_ROUNDS: usize = 1000;
```
Comment (nategraf) [inline risc0/zkvm/platform/src/syscall.rs:76]: `MAX_SHA_COMPRESS_ROUNDS` might be misread to imply that it controls the number of rounds within the compression function.
```suggestion
pub const MAX_SHA_COMPRESS_BLOCKS: usize = 1000;
```

Comment (nategraf) [inline risc0/zkvm/platform/src/syscall.rs:76]: Do we also need to add a check to the build for the `Executor` that the segment limit is not set low enough that the max number of read or compress invocations does not exceed the segment limit?

Comment (flaub) [reply risc0/zkvm/platform/src/syscall.rs:76]: WDYT of `MAX_SHA_COMPRESS_COUNT`? I can also add a comment

Comment (flaub) [reply risc0/zkvm/platform/src/syscall.rs:76]: Good call, I'll review what the new minimum needs to be based on these parameters.

File: risc0/zkvm/platform/src/syscall.rs [hunk 2]
```diff
@@ -277,15 +280,20 @@ pub unsafe extern "C" fn sys_sha_buffer(
...
-        );
+        let mut count_remain = count;
+        while count_remain > 0 {
+            let count = min(count_remain, MAX_SHA_COMPRESS_ROUNDS as u32);
+            asm!(
+                "ecall",
+                in("t0") ecall::SHA,
+                in("a0") out_state,
+                in("a1") in_state,
+                in("a2") buf,
```
[the remark this answers is not in the document]
Comment (flaub) [reply risc0/zkvm/platform/src/syscall.rs:291]: Good find! I'll fix this and add a test

**Comment by flaub** [comment]
@nategraf Hopefully addressed all your comments, thank you!

**Review by jbruestle — APPROVED** [review]
Looks good.  Notable that we will need to pull into WIP and since the changes to the set of PO2s will modify things like the recursion STARK->SNARK final hashes, etc.

File: risc0/zkp/src/lib.rs [hunk 3]
```diff
@@ -44,7 +44,7 @@ pub mod hal {
 
 pub use risc0_core::field;
 
-pub const MIN_CYCLES_PO2: usize = 11;
+pub const MIN_CYCLES_PO2: usize = 13;
 pub const MIN_CYCLES: usize = 1 << MIN_CYCLES_PO2; // 1K
```
Comment (nategraf) [inline risc0/zkp/src/lib.rs:48]:
```suggestion
pub const MIN_CYCLES: usize = 1 << MIN_CYCLES_PO2; // 8K
```

**Review by nategraf — APPROVED** [review]

**Pull request merged**